### DISCUSSION TITLE
[Feature Request] Add a `/skyblock/player` endpoint

### DISCUSSION THREAD

**Discussion by ptlthg** [HypixelDev/PublicAPI#669] [Ideas]
This new endpoint could be the home of a lot of useful data related to a SkyBlock player, without needing to request more data than needed.

 - Basic Hypixel player information (ex: id, ign, rank, social media)
 	- These should be included to avoid needing to call the `/player` endpoint at all
 	- In the example response, I included formatted names with color codes. This isn't necessary as long as rank info is provided, but it would be nice.
 - List of profile details (ex: id, cute_name, game_mode, members, api)
 	- I've included a museum api flag here, this way we know if a request to the museum api would be a waste of a request or not beforehand.
 	- I also want ranks of profile members to be included here, which would remove the requirement of websites fetching the `/player` endpoint for every profile member for the purpose of showing ranks.
 - `community_upgrades` (account upgrades)
 	- Currently not available anywhere in the API, if there are issues with including this read #647 for my idea there.
 - Data like claimed Scorpius bribes, and `skyblock_extra`
 
 Here's what I imagine a response would look like:
 ```json
{
  "success": true,
  "player": {
    "id": "7da0c47581dc42b4962118f8049147b7",
    "name": "Ke5o",
    "formatted_name": "§b[MVP§4+§b] Ke5o",
    "socialMedia": {
      "DISCORD": "kaeso.dev"
    },
    "profiles": {
      "7da0c47581dc42b4962118f8049147b7": {
        "profile_id": "7da0c47581dc42b4962118f8049147b7",
        "cute_name": "Raspberry",
        "game_mode": "island",
        "members": [
          {
            "id": "7da0c47581dc42b4962118f8049147b7",
            "name": "Ke5o",
            "formatted_name": "§b[MVP§4+§b] Ke5o",
            "api": {
              "museum": false
            }
          }
        ],
      }
    },
    "community_upgrades": {
      "currently_upgrading": [],
      "upgrade_states": [
        {
          "upgrade": "garden_fortune",
          "tier": 1,
          "started_ms": 1603242493693,
          "started_by": "7da0c47581dc42b4962118f8049147b7",
          "claimed_ms": 1605420029641,
          "claimed_by": "7da0c47581dc42b4962118f8049147b7",
          "fasttracked": false
        }
      ]
    },
    "claimed_rewards": {
      "claimed_potato_talisman": 1592794465850,
      "skyblock_free_cookie": 1599596098033,
      "scorpius_bribe_96": 1603235712261,
      "claimed_century_cake": 1604512198402
    },
    "extra": {
      "ozanne_coins": 5000000
    }
  }
}
```

I'm sure this endpoint could also be used for more than what I showed above, but I think there's a clear need for this endpoint to avoid including random SkyBlock data in `/player`, and to reduce requests sent.

 **EDIT:** Someone totally didn't leak what data Hypixel has for overall SkyBlock player data, so I would also like these properties below exposed:
 
 - `selected_profile` 
   - Can be used to only request the one profile for someone instead of all profiles
 - `elections`
   - Would just be cool to include, but it might encourage people to scrape this, so I wouldn't mind it being left unexposed.
 - `cookies_eaten`
   - I don't think this would expose monetization info because the bazaar exists, but it's not a big deal to exclude it.
 - Misc things that would be fun
   - `one_time_rewards_claimed`
   - `collected_hat_years`
   - `collected_cake_years`

**Comment by DuckySoLucky** [comment]
> EDIT: Someone totally didn't leak what data Hypixel has for overall SkyBlock player data, so I would also like these properties below exposed:

*for legal reasons this was a joke*
